addons/web/static/tests/fields/relational_fields_tests.js

```diff
@@ -3206,6 +3206,169 @@ QUnit.module('relational_fields', {
+
+        form.destroy();
+    });
+
+    QUnit.test('onchange for embedded one2many in a one2many updated by server', function (assert) {
+        //here we test that after an onchange, the embedded one2many field has
+        //been updated by a new list of ids by the server response, to this new
+        //list should be correctly sent back at save time
+        assert.expect(2);
+
+        var numUserOnchange = 0;
+
+        this.data.turtle.fields.partner_ids.type = "one2many";
+        this.data.partner.records[0].turtles = [2];
+        this.data.turtle.records[1].partner_ids = [2];
+
+        this.data.partner.onchanges = {
+            turtles: function (obj) {
+                if (numUserOnchange === 0){
+                obj.turtles = [
+                      [5],
+                      [1, 2,
+                        {
+                          "turtle_foo": "hop",
+                          "partner_ids": [[5], [4, 2], [4, 4]]
+                        }
+                      ]
+                ]
+                }
+                numUserOnchange++;
+            }
+        };
+
+        var form = createView({
+            View: FormView,
+            model: 'partner',
+            data: this.data,
+            arch:'<form string="Partners">' +
+                    '<sheet>' +
+                        '<notebook>' +
+                            '<page string="P page">' +
+                                '<field name="turtles">' +
+                                    '<tree editable="bottom">' +
+                                        '<field name="turtle_foo"/>' +
+                                        '<field name="partner_ids" widget="many2many_tags"/>' +
+                                    '</tree>' +
+                                '</field>' +
+                            '</page>' +
+                        '</notebook>' +
+                    '</sheet>' +
+                 '</form>',
+            res_id: 1,
+            mockRPC: function (route, args) {
+                if (route === '/web/dataset/call_kw/partner/onchange') {
+                }
+                if (route === '/web/dataset/call_kw/partner/write') {
+                    var expected_result_turtles = [[1,2,
+                        {"partner_ids":[[4,2,false],[4,4,false]],
+                         "turtle_foo":"hop"}]
+                    ];
+
+                    assert.deepEqual(args.args[1].turtles, expected_result_turtles,
+                                     'The right values should be written')
+                    console.log("WRITE");
+                    console.log(arguments);
+                    debugger;
+                }
+                return this._super(route, args);
+            }
+        });
+
+        form.$buttons.find('.o_form_button_edit').click();
+        form.$('.o_data_cell').eq(1).click();
+        var cell = form.$('.o_selected_row .o_input[name=turtle_foo]')
```

Comment:
check that there is only one record in the sub O2M before triggering the onchange, and check that there a two after?